### PR TITLE
Add GC in train.py by defalut and add download script for dataset.

### PR DESCRIPTION
**Pull request by qingqing01**

### CODE REVIEW

File: PaddleCV/object_detection/tools/train.py [hunk 1]
```diff
@@ -19,9 +19,22 @@
...
+    for key, value in flags.items():
+        if os.environ.get(key, None) is None:
+            os.environ[key] = str(value)
+
+
+# NOTE(paddle-dev): All of these flags should be
+# set before `import paddle`. Otherwise, it would
+# not take any effect. 
+set_paddle_flags({
+    'FLAGS_eager_delete_tensor_gb': 0,  # enable GC to save memory
```
Comment (heavengate) [inline PaddleCV/object_detection/tools/train.py]: export FLAGS_fast_eager_deletion_mode=1  这个需要设置么

File: PaddleCV/object_detection/docs/INSTALL.md [hunk 2]
```diff
@@ -99,34 +99,23 @@ ln -sf <path/to/voc> $PaddleDetection/data/voc
...
 ```
 
 **Auto download datasets:**
 
 If you set up models while `data/coc` and `data/voc` is not found, PaddleDetection will automaticaly download them from [MSCOCO-2017](http://images.cocodataset.org) and [VOC2012](http://host.robots.ox.ac.uk/pascal/VOC), the decompressed datasets will be places in `~/.cache/paddle/dataset/` and can be discovered automaticaly in the next setting up time.
 
 
-**NOTE:** For further informations on the datasets, please see [DATASET.md](../ppdet/data/README.md)
-
+**NOTE:** For further informations on the datasets, please see [DATASET.md](docs/DATASET.md)
```
Comment (heavengate) [inline PaddleCV/object_detection/docs/INSTALL.md]: docs/DATASET.md  -> ./DATASET.md

File: PaddleCV/object_detection/tools/train.py (hunk 1 above)
Comment (qingqing01) [reply PaddleCV/object_detection/tools/train.py]: 这个参数默认是1

File: PaddleCV/object_detection/dataset/coco/download.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
+DIR="$( cd "$(dirname "$0")" ; pwd -P )"
```
Comment (willthefrog) [inline PaddleCV/object_detection/dataset/coco/download.sh]: may need to update `INSTALL.md` to reflect this?

File: PaddleCV/object_detection/tools/eval.py [hunk 4]
```diff
@@ -96,11 +95,10 @@ def main():
 
     keys, values, cls = parse_fetches(fetches, eval_prog, extra_keys)
 
-    # 6. Run
+    # do evaluation
```
Comment (willthefrog) [inline PaddleCV/object_detection/tools/eval.py]: simply `evaluation` should be fine

File: PaddleCV/object_detection/tools/train.py [hunk 5]
```diff
@@ -19,9 +19,22 @@
 import os
 import time
 import multiprocessing
-
 import numpy as np
 
+
+def set_paddle_flags(flags):
```
Comment (willthefrog) [inline PaddleCV/object_detection/tools/train.py]: simplify with `**kwargs`?

File: PaddleCV/object_detection/tools/train.py [hunk 6]
```diff
@@ -151,10 +165,9 @@ def main():
             checkpoint.save(exe, train_prog, os.path.join(save_dir, str(it)))
 
             if FLAGS.eval:
-                # Run evaluation
+                # do evaluation
```
Comment (willthefrog) [inline PaddleCV/object_detection/tools/train.py]: just `evaluation` should be fine

File: PaddleCV/object_detection/docs/INSTALL.md (hunk 2 above)
Comment (qingqing01) [reply PaddleCV/object_detection/docs/INSTALL.md]: Done

File: PaddleCV/object_detection/tools/train.py (hunk 6 above)
Comment (qingqing01) [reply PaddleCV/object_detection/tools/train.py]: Done

File: PaddleCV/object_detection/tools/train.py (hunk 5 above)
Comment (qingqing01) [reply PaddleCV/object_detection/tools/train.py]: Done

**Review by heavengate — APPROVED** [review]
LGTM

**Review by willthefrog — APPROVED** [review]
LGTM.  you can merge it any time. I will rebase and revise the docs in my PR.

**Pull request merged**